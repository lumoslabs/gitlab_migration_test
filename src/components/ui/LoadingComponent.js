

import React from 'react';
import Lottie from 'lottie-react';
import { css, StyleSheet } from 'aphrodite';
import lumositySpinner from '../assets/loader_spin_black.json';
import { base } from '../styles/colors';
import commonStyles from '../styles/commonStyles';

const LoadingComponent = (props) => {
  return (
    <div className={css([commonStyles.flexColumnAllCenter, styles.loadingAnimdiv])}>
      {props.title &&
        <div className={css([commonStyles.flexColumnAllCenter, styles({
          titleDivHeight: props.titleDivHeight ? props.titleDivHeight : '20vh'
          }).titleDiv])}
        >
          <p className={css(styles.loadingTitle)}>
            {props.title}
          </p>
        </div>
      }
      <div className={css([commonStyles.flexColumnAllCenter, styles({
        loadingDivHeight: props.loadingDivHeight ? props.loadingDivHeight : '40vh'
        }).loadingdiv]) }
      >
        <Lottie animationData={lumositySpinner} />
      </div>
    </div>
  );
};

const styles = (props) => StyleSheet.create({
  loadingAnimdiv: {
    display: 'flex',
    flexDirection: 'column'
  },

  titleDiv: {
    height: props.titleDivHeight
  },
  loadingTitle: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '1.95vw',
    fontWeight: 500
  },

  loadingdiv: {
    height: props.loadingDivHeight
  },
});

export default LoadingComponent;
