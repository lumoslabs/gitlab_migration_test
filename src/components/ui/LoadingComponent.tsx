import React from 'react'
import { css, StyleSheet } from 'aphrodite'
import base from '@styles/colors/base'
import commonStyles from '@styles/commonStyles'

const baseStyles = StyleSheet.create({
  loadingAnimdiv: {
    display: 'flex',
    flexDirection: 'column'
  },
  loadingTitle: {
    color: base.lumosBlack,
    fontFamily: 'MuseoSans500',
    fontSize: '1.95vw',
    fontWeight: 500
  },
})

const customStyles = (titleDivHeight, loadingDivHeight) => StyleSheet.create({
  titleDiv: {
    height: titleDivHeight
  },
  loadingdiv: {
    height: loadingDivHeight
  },
})

export interface ILoadingComponent {
  title?: string;
  titleDivHeight?: string;
  loadingDivHeight?: string;
}

const LoadingComponent = ({ title, titleDivHeight = '20vh', loadingDivHeight  = '40vh' }: ILoadingComponent): JSX.Element => {
  const customizedStyles = customStyles(titleDivHeight, loadingDivHeight)

  return (
    <div className={css([commonStyles.flexColumnAllCenter, baseStyles.loadingAnimdiv])}>
      {title && (
        <div className={css([commonStyles.flexColumnAllCenter, customizedStyles.titleDiv])}>
          <p className={css(baseStyles.loadingTitle)}>
            {title}
          </p>
        </div>
      )}
      <div className={css([commonStyles.flexColumnAllCenter, customizedStyles.loadingdiv])}>
        <div className='loader' />
      </div>
    </div>
  )
}

export default LoadingComponent
