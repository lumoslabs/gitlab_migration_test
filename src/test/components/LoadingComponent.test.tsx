import React from 'react'
import LoadingComponent from '@components/ui/LoadingComponent'
import { render } from '@testing-library/react'

describe('components', () => {
  describe('LoadingComponent', () => {
    it('matches snapshot', () => {
      const { asFragment } = render(
        <LoadingComponent />, {}
      )
      expect(asFragment()).toMatchSnapshot()
    })
  })
})
