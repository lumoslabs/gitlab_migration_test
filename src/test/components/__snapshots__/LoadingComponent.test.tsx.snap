// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components LoadingComponent matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="flexColumnJustifyCenter_66qnx8-o_O-fullHeight_149btvt"
  >
    <div
      class="loader"
    />
  </div>
</DocumentFragment>
`;
